pkg/go/graph/weighted_graph_builder_test.go

```diff
@@ -0,0 +1,384 @@
+package graph
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/stretchr/testify/require"
+
+	language "github.com/openfga/language/pkg/go/transformer"
+)
+
+func TestWeightedGraphBuilder(t *testing.T) {
+	t.Parallel()
+
+	testCases := map[string]struct {
+		model          string
+		expectedOutput string // can visualize in https://dreampuf.github.io/GraphvizOnline
+		expectedError  error
+	}{
+		`direct_assignment_to_one_type`: {
+			model: `
+				model
+					schema 1.1
+				type user
+				type folder
+					relations
+						define viewer: [user]`,
+			expectedOutput: `digraph {
+graph [
+rankdir=TB
+];
+
+// Node definitions.
+0 [label=folder];
+1 [label="folder#viewer - weights:[user=1]"];
+2 [label=user];
+
+// Edge definitions.
+1 -> 2 [label="direct - weights:[user=1]"];
+}`,
+		},
+		`direct_assignment_to_multiple_types`: {
+			model: `
+				model
+					schema 1.1
+				type user1
+				type user2
+				type folder
+					relations
+						define viewer: [user1, user2]`,
+			expectedOutput: `digraph {
+graph [
+rankdir=TB
+];
+
+// Node definitions.
+0 [label=folder];
+1 [label="folder#viewer - weights:[user1=1,user2=1]"];
+2 [label=user1];
+3 [label=user2];
+
+// Edge definitions.
+1 -> 2 [label="direct - weights:[user1=1]"];
+1 -> 3 [label="direct - weights:[user2=1]"];
+}`,
+		},
+		`direct_assignment_to_type_and_wildcard`: {
+			model: `
+				model
+				  schema 1.1
+				
+				type user
+				
+				type folder
+				  relations
+					define viewer: [user,user:*]
+	`,
+			expectedOutput: `digraph {
+graph [
+rankdir=TB
+];
+
+// Node definitions.
+0 [label=folder];
+1 [label="folder#viewer - weights:[user=1]"];
+2 [label=user];
+3 [label="user:*"];
+
+// Edge definitions.
+1 -> 2 [label="direct - weights:[user=1]"];
+1 -> 3 [label=direct];
```

Comment:
Is there any weight to this edge?

Reply:
No, `type:*` nodes don't have any weights, therefore any edges in their direction don't have weights either